src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -55,24 +57,65 @@ public override bool Execute()
                 {
                     Log.LogError("Missing required parameter.  BuildNumberOverrideStructureRegex must be provided if PackageBuildNumberOverride is provided.");
                 }
-                buildNumberOverrideStructureRegex = new Regex(BuildNumberOverrideStructureRegex);
-                if(!buildNumberOverrideStructureRegex.IsMatch(PackageBuildNumberOverride))
+                _buildNumberOverrideStructureRegex = new Regex(BuildNumberOverrideStructureRegex);
+                if(!_buildNumberOverrideStructureRegex.IsMatch(PackageBuildNumberOverride))
                 {
                     Log.LogError("Invalid package version format: '{0}'", PackageBuildNumberOverride);
                     return false;
                 }
             }
 
             JObject projectRoot = ReadProject(ProjectJson);
+            var invalidFramework = AreValidFrameworkPaths(projectRoot);
+            if(invalidFramework != string.Empty)
+            {
+                OutputProjectJson = ProjectJson;
+                Log.LogError("Unable to find framework section '{0}' in '{1}'", invalidFramework, ProjectJson);
+                return false;
+            }
+            _versionStructureRegex = new Regex(VersionStructureRegex);
+
+            // No PackageBuildNumberOverride was specified, so try to find one to associate with our AdditionalDependencies
+            PackageBuildNumberOverride = PackageBuildNumberOverride ?? DeriveBuildNumber(projectRoot);
 
-            versionStructureRegex = new Regex(VersionStructureRegex);
-            JObject dependencies = GenerateDependencies(projectRoot);
-            projectRoot = UpdateProperty("dependencies", projectRoot, dependencies);
+            if (Frameworks == null || Frameworks.Count() == 0)
```

Comment:
Frameworks.Length == 0. In general perfer concrete Lenght or Count properties over the Count() extension method.